Build of whole project is broken.

Using command from README.md:
```
mvn clean install -Pqulice
```

I tried to build repo on my local machine (Windows 10) and it failed with:
```
[INFO]
[INFO] --- jacoco-maven-plugin:0.7.8:check (jacoco-check) @ jpeek ---
[INFO] Loading execution data file C:\Users\Nikita_Puzankov\workspace\personal\jpeek\target\jacoco.exec
[INFO] Analyzed bundle 'jpeek' with 35 classes
[WARNING] Rule violated for bundle jpeek: lines covered ratio is 0.65, but expected minimum is 0.67
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 03:58 min
[INFO] Finished at: 2019-04-19T22:13:29+04:00
[INFO] Final Memory: 33M/329M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.jacoco:jacoco-maven-plugin:0.7.8:check (jacoco-check) on project jpeek: Coverage checks have not been met. See log for details. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```
**Without any local changes.**

I would like to see working build to work on the project instead of covering code with tests for free :-)

--- a/README.md
+++ b/README.md
@@ -257,6 +257,7 @@ if lost.
   - [@llorllale](https://github.com/llorllale) as George Aristy
   - [@mesut](https://github.com/mesut) as Mesut Özen
   - [@serranya](https://github.com/serranya) as Peter Lamby
+  - [@humb1t](https://github.com/humb1t) as Nikita Puzankov
 
 Don't hesitate to add your name to this list in your next pull request.
 

--- a/src/main/java/org/jpeek/Main.java
+++ b/src/main/java/org/jpeek/Main.java
@@ -125,12 +125,31 @@ public final class Main {
      * @throws IOException If fails
      */
     private void run() throws IOException {
-        final ConsoleAppender console = new ConsoleAppender();
-        if (!this.quiet) {
-            console.setLayout(new PatternLayout("%m%n"));
-            console.activateOptions();
-            Logger.getRootLogger().addAppender(console);
+        if (this.overwrite && this.sources.equals(this.target)) {
+            throw new IllegalArgumentException(
+                "Invalid paths - can't be equal if overwrite option is set."
+            );
         }
+        final ConsoleAppender console = this.buildConsoleAppender();
+        final Map<String, Object> params = this.buildParameters();
+        new App(
+            this.sources.toPath(),
+            new FileTarget(
+                this.target,
+                this.overwrite
+            ).toPath(),
+            params
+        ).analyze();
+        if (!this.quiet) {
+            Logger.getRootLogger().removeAppender(console);
+        }
+    }
+
+    /**
+     * Prepare application parameters based on configuration and metrics.
+     * @return A {@link Map} filled with parameters.
+     */
+    private Map<String, Object> buildParameters() {
         final Map<String, Object> params = new HashMap<>(0);
         if (this.ctors) {
             params.put("include-ctors", 1);
@@ -149,16 +168,20 @@ public final class Main {
             }
             params.put(metric, true);
         }
-        new App(
-            this.sources.toPath(),
-            new FileTarget(
-                this.target,
-                this.overwrite
-            ).toPath(),
-            params
-        ).analyze();
+        return params;
+    }
+
+    /**
+     * Prepare {@link ConsoleAppender} based on configuration.
+     * @return Configured {@link ConsoleAppender}.
+     */
+    private ConsoleAppender buildConsoleAppender() {
+        final ConsoleAppender console = new ConsoleAppender();
         if (!this.quiet) {
-            Logger.getRootLogger().removeAppender(console);
+            console.setLayout(new PatternLayout("%m%n"));
+            console.activateOptions();
+            Logger.getRootLogger().addAppender(console);
         }
+        return console;
     }
 }

--- a/src/test/java/org/jpeek/MainTest.java
+++ b/src/test/java/org/jpeek/MainTest.java
@@ -96,7 +96,7 @@ public final class MainTest {
     public void crashesIfMetricsHaveInvalidNames() throws IOException {
         final Path target = Files.createTempDirectory("");
         new Assertion(
-        "Must throw an exception",
+            "Must throw an exception",
             () -> {
                 Main.main(
                     "--sources", Paths.get(".").toString(),

--- a/src/test/java/org/jpeek/MainTest.java
+++ b/src/test/java/org/jpeek/MainTest.java
@@ -31,6 +31,8 @@ import java.nio.file.Paths;
 import org.hamcrest.MatcherAssert;
 import org.hamcrest.Matchers;
 import org.junit.Test;
+import org.llorllale.cactoos.matchers.Assertion;
+import org.llorllale.cactoos.matchers.Throws;
 
 /**
  * Test case for {@link Main}.
@@ -65,6 +67,49 @@ public final class MainTest {
             "--sources", Paths.get(".").toString(),
             "--target", target.toString()
         );
+    }
+
+    @Test
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    public void crashesIfOverwriteAndSourceEqualsToTarget() throws IOException {
+        final Path source = Files.createTempDirectory("sourceequalstarget");
+        final Path target = source;
+        new Assertion(
+            "Must throw an exception",
+            () -> {
+                Main.main(
+                    "--sources", source.toString(),
+                    "--target", target.toString(),
+                    "--overwrite"
+                );
+                return true;
+            },
+            new Throws(
+                "Invalid paths - can't be equal if overwrite option is set.",
+                IllegalArgumentException.class
+            )
+        ).affirm();
+    }
+
+    @Test
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    public void crashesIfMetricsHaveInvalidNames() throws IOException {
+        final Path target = Files.createTempDirectory("");
+        new Assertion(
+        "Must throw an exception",
+            () -> {
+                Main.main(
+                    "--sources", Paths.get(".").toString(),
+                    "--target", target.toString(),
+                    "--metrics", "#%$!"
+                );
+                return true;
+            },
+            new Throws(
+                "Invalid metric name: '#%$!'",
+                IllegalArgumentException.class
+            )
+        ).affirm();
     }
 
     @Test

--- a/src/test/java/org/jpeek/web/DyNumTest.java
+++ b/src/test/java/org/jpeek/web/DyNumTest.java
@@ -1,0 +1,103 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2019 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.jpeek.web;
+
+import org.hamcrest.core.IsEqual;
+import org.hamcrest.core.IsNot;
+import org.hamcrest.core.IsNull;
+import org.junit.Test;
+import org.llorllale.cactoos.matchers.Assertion;
+
+/**
+ * Test case for {@link DyNum}.
+ * @author Nikita Puzankov (humb1t@yandex.ru)
+ * @version $Id$
+ * @since 0.31
+ * @checkstyle JavadocMethodCheck (500 lines)
+ */
+public final class DyNumTest {
+
+    @Test
+    public void testConstructDyNumFromString() {
+        final double number = 2.0d;
+        new Assertion<>(
+            "Double values should be equal",
+            new DyNum(
+                String.valueOf(number)
+            )::doubleValue,
+            new IsEqual<>(number)
+        ).affirm();
+    }
+
+    @Test
+    public void testConstructDyNumFromLong() {
+        final long number = 2L;
+        new Assertion<>(
+            "Long values should be equal",
+            new DyNum(number)::longValue,
+            new IsEqual<>(number)
+        ).affirm();
+    }
+
+    @Test
+    public void testConstructDyNumFromLongCompareToInt() {
+        final Double number = 2.0d;
+        new Assertion<>(
+            "Integer values should be equal",
+            new DyNum(number)::intValue,
+            new IsEqual<>(number.intValue())
+        ).affirm();
+    }
+
+    @Test
+    public void testConstructDyNumFromLongCompareToFloat() {
+        final float number = 2.0f;
+        new Assertion<>(
+            "Float values should be equal",
+            new DyNum(number)::floatValue,
+            new IsEqual<>(number)
+        ).affirm();
+    }
+
+    @Test
+    public void testConstructDyNumFromDouble() {
+        final double number = 2.0d;
+        new Assertion<>(
+            "Double values should be equal.",
+            new DyNum(number)::doubleValue,
+            new IsEqual<>(number)
+        ).affirm();
+    }
+
+    @Test
+    public void testUpdateDyNum() {
+        new Assertion<>(
+            "Result should not be null",
+            new DyNum(2.0d)::update,
+            new IsNot<>(
+                new IsNull<>()
+            )
+        ).affirm();
+    }
+}

--- a/src/test/java/org/jpeek/web/PagesTest.java
+++ b/src/test/java/org/jpeek/web/PagesTest.java
@@ -53,7 +53,7 @@ public final class PagesTest {
             StandardOpenOption.CREATE_NEW
         );
         new Assertion<>(
-            "should return -1",
+            "There is no byte available in body - should return -1",
             () -> new Pages(temp).apply(path).body().read(),
             new IsEqual<>(-1)
         ).affirm();

--- a/src/test/java/org/jpeek/web/PagesTest.java
+++ b/src/test/java/org/jpeek/web/PagesTest.java
@@ -1,0 +1,61 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2019 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.jpeek.web;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import org.hamcrest.core.IsEqual;
+import org.junit.Test;
+import org.llorllale.cactoos.matchers.Assertion;
+
+/**
+ * Test case for {@link Pages}.
+ * @author Nikita Puzankov (humb1t@yandex.ru)
+ * @version $Id$
+ * @since 0.31
+ * @checkstyle JavadocMethodCheck (500 lines)
+ */
+public final class PagesTest {
+
+    /**
+     * Simple test.
+     */
+    @Test
+    public void testApply() throws IOException {
+        final Path temp = Files.createTempDirectory("");
+        temp.resolve("a").toFile().mkdirs();
+        final String path = "a/z.class";
+        Files.write(
+            temp.resolve(path), "".getBytes(),
+            StandardOpenOption.CREATE_NEW
+        );
+        new Assertion<>(
+            "should return -1",
+            () -> new Pages(temp).apply(path).body().read(),
+            new IsEqual<>(-1)
+        ).affirm();
+    }
+}

